ece.ts

```diff
@@ -180,16 +180,32 @@ export function unpad(record: ArrayBuffer): ArrayBuffer {
 /**
  * @param data is the encrypted data with the header block
  * @param secret is the secret used to encrypt the data
- * @param header an optional header if not part of data.
+ * @param headerOrOptions an optional header if not part of data, or options object 
  * @returns the decrypted data
  */
 export async function decrypt(
   data: ArrayBuffer,
   secret: ArrayBuffer,
-  header?: Header,
+  headerOrOptions?: Header | ECECryptoOptions,
 ): Promise<ArrayBuffer> {
-  if (header === undefined) header = Header.fromBytes(data);
-  const crypto = new ECECrypto(secret, { header });
+  let options: ECECryptoOptions;
+  let header: Header;
+  if (headerOrOptions instanceof Header) {
+    header = headerOrOptions;
+    options = { header: headerOrOptions };
+  } else if (headerOrOptions) {
+    if (headerOrOptions.header instanceof Header) {
+      header = headerOrOptions.header;
+    } else {
+      header = new Header(headerOrOptions.header || {});
+    }
+    options = { ...headerOrOptions, header };
+  } else {
+    header = Header.fromBytes(data);
+    options = { header };
+  }
```

Comment:
This seems too complex. Why would this not work ?

```ts
  let options: ECECryptoOptions;
  if (headerOrOptions) {
    if (headerOrOptions instanceof Header) options = { header: headerOrOptions };
    else options = headerOrOptions
  } else {
      options = { header: Header.fromBytes(data) }
  }
```

Reply:
Because further down it needs a `Header` instance whereas the `header` in the options might just be the options for a header not a `Header` itself

Reply:
although given `header` is public in `ECECrypto` we could also read it from there

Reply:
Yeah that's the best solution IMO

Reply:
Done! On phone though now so haven't tested it

Reply:
line 195:
-    options = headerOrOptions || {};
+    options = headerOrOptions || { header: Header.fromBytes(data) };

Reply:
I can do the changes myself if you can't.

Reply:
Ah yeh. I just pushed something that I think solves that? Always default to reading from bytes if it's missing in options or there are no options